### DISCUSSION TITLE
Swift Compiler Error (Xcode): Expected declaration - FirebaseAuth

### DISCUSSION THREAD

**Discussion by MohamedYassineDeneubourg** [firebase/flutterfire#13505] [Q&A]
Hello i have this issue with Flutter..

[✓] Flutter (Channel stable, 3.19.6, on macOS 12.7.6 21H1320 darwin-x64, locale fr-BE)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.94.1)
[✓] Connected device (3 available)
[✓] Network resources

ERROR here:

Launching lib/main.dart on iPhone 14 Pro in debug mode...
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Running pod install...                                             48,0s
Running Xcode build...                                                  
Xcode build done.                                           132,3s
Failed to build iOS app
Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/Auth/
Auth.swift:287:2

Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/AuthP
rovider/OAuthProvider.swift:42:2

Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/AuthP
rovider/OAuthProvider.swift:68:2

Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/AuthP
rovider/OAuthProvider.swift:146:2

Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/AuthP
rovider/OAuthProvider.swift:183:2

Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/AuthP
rovider/OAuthProvider.swift:213:2

Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/AuthP
rovider/OAuthProvider.swift:238:2

Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/User/
User.swift:98:2

Swift Compiler Error (Xcode): Expected declaration
/Users/yassine/Desktop/Nafsy-App/nafsy_app/ios/Pods/FirebaseAuth/FirebaseAuth/Sources/Swift/User/
User.swift:140:2

Could not build the application for the simulator.
Error launching application on iPhone 14 Pro.

Can someone help me?

**Comment by ctrlVnt** [comment]
I had the same problem and i found it:

![immagine](https://github.com/user-attachments/assets/ee222b33-23a1-4660-ad92-0debb7020e7c)

If you are using Firebase you need more recent version of Xcode, the 15.2

So you need Ventura minimum
